app/api/attest/route.ts

```diff
@@ -1,67 +1,88 @@
+"use server";
+
 import { type NextRequest, NextResponse } from "next/server";
 import { JsonRpcProvider, Wallet } from "ethers";
 import { auth } from "@/lib/auth";
 import { EIP712Proxy } from "@ethereum-attestation-service/eas-sdk/dist/eip712-proxy";
-import { isDiamondHands } from "@/lib/diamond-hands"
-import { jsonStringifyBigInt } from "@/lib/utils"
+import { isDiamondHands } from "@/lib/diamond-hands";
+import { jsonStringifyBigInt } from "@/lib/utils";
+import { getRedisInstance } from "@/lib/redis";
+import { ATTESTATION_DEADLINE } from "@/lib/config";
 
 import {
   PROXY_CONTRACT_ADDRESS,
   PRIVATE_KEY,
   DIAMOND_HANDS_SCHEMA_UID,
-  DIAMOND_HANDS_ATTESTATION_DATA
-} from "@/lib/config"
-
+  DIAMOND_HANDS_ATTESTATION_DATA,
+} from "@/lib/config";
 
 if (!PROXY_CONTRACT_ADDRESS) {
-  throw new Error('PROXY_CONTRACT_ADDRESS is not set')
+  throw new Error("PROXY_CONTRACT_ADDRESS is not set");
 }
 
 if (!PRIVATE_KEY) {
-  throw new Error('PRIVATE_KEY is not set')
+  throw new Error("PRIVATE_KEY is not set");
 }
 
-export const runtime = "edge";
+//export const runtime = "edge";
```

Comment:
I needed to comment this to get the code to run, I'm not sure what edge runtime actually does but will look into it.

Reply:
It specifies that this API route runs in a runtime that is not node.js and thus can be optimized to run geographically closer to the user (like a CDN).

It s not technically necessary, mostly a performance optimization.

Reply:
I guess if we want to use Vercel's redis compatible KV store, we need to do it like this: https://github.com/riccardogiorato/redis-with-vercel-edge/blob/main/src/pages/api/redis.ts